pkgs/tools/networking/wireguard-tools/default.nix

```diff
@@ -8,7 +8,7 @@ stdenv.mkDerivation rec {
 
   src = fetchzip {
     url = "https://git.zx2c4.com/WireGuard/snapshot/WireGuard-${version}.tar.xz";
-    sha256 = "ff653095cc0e4c491ab6cd095ddf5d1db207f48f947fb92873a73220363f423c";
+    sha256 = "0944zxmpx2cs71nxl7rcyhpqlwplkzd7jsf1n66vflngw2sjxm03";
```

Comment:
The checksum is calculated over the content of the zip archive as it was unclear whether cgit might regenerate archives resulting in a different hash. `nix-prefetch-url --unpack url` can be used to generate those.

Reply:
Interesting. I didn't know Nix could do this. How does it know which type of checksum it is? Does it just trial check all possibilities? Or does that hex string have a type built in somewhere?

Reply:
It depends on the fetcher used. The default one is `fetchurl`. Here `fetchzip` is used.

Reply:
Neat.